### PR TITLE
contrib: use `keys.openpgp.org` to retrieve builder keys

### PR DESCRIPTION
**Pull request by fanquake**
`hkps://hkps.pool.sks-keyservers.net` is essentially no-longer functional,
and a number of distributions and GPG tools have since switched to using
the `keys.openpgp.org` key server as their default.

See this Debian patch for additional context:
https://salsa.debian.org/debian/gnupg2/-/blob/debian/main/debian/patches/Use-hkps-keys.openpgp.org-as-the-default-keyserver.patch

Switch to using keys.openpgp.org in the CI as well.

### CODE REVIEW

**Comment by hebasto** [comment]
Concept ACK.

Does `gpg --refresh-keys` also require specifying of a key server?

**Comment by Zero-1729** [comment]
Concept ACK

@hebasto I don't think it should, but maybe as a precaution (just in case there are any revoked keys, new user IDs, sigs, etc. or if facing remote server issues). Just spitballing here though.

<details>
    <summary>fetch_keys.sh</summary>
    
    #!/bin/zsh                                                                                                                                      
 
    while read fingerprint keyholder_name
        do gpg --keyserver hkp://subset.pool.sks-keyservers.net --recv-keys ${fingerprint}
    done < ./keys.txt
</details>

Also, I tried to run the script above with `subset.pool.sks-keyservers.net`, and it just returned `gpg: keyserver receive failed: No name` for all entries in `contrib/builder-keys/keys.txt`. It only successfully fetched the keys in the file after swapping out the keyserver with `keyserver.ubuntu.com:80`.

**Comment by Zero-1729** [comment]
Is SKS being deprecated or something? Found a related [StackExchange question](https://unix.stackexchange.com/questions/656205/sks-keyservers-gone-what-to-use-instead) which referenced the following message from 'https://sks-keyservers.net/ ':

```
This service is deprecated. This means it is no longer maintained, and new HKPS certificates will not be issued. Service reliability should not be expected.

Update 2021-06-21: Due to even more GDPR takedown requests, the DNS records for the pool will no longer be provided at all.
```

**Comment by jamesob** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/22688/commits/641155aef9583134bbb52cd97e83f5c76561169c

**Comment by dongcarl** [comment]
Just a datapoint: it would seem that since the launch of keys.opengpg.org, many GPG tools and certain distros have switched over to that as the default server.

Debian switched in July 2019: https://salsa.debian.org/debian/gnupg2/-/blob/01898735a015541e3ffb43c7245ac1e612f40836/debian/NEWS#L10-29

Reproduced:
```
  Upstream GnuPG now defaults to not accepting third-party certifications
  from the keyserver network.  Given that the SKS keyserver network is
  under attack via certificate flooding, and third-party certifications
  will not be accepted anyway, we now ship with the more tightly-constrained
  and abuse-resistant system hkps://keys.openpgp.org as the default
  keyserver.

  Users with bandwidth to spare who want to try their luck with the SKS
  pool should add the following line to ~/.gnupg/dirmngr.conf to revert to
  upstream's default keyserver:

      keyserver hkps://hkps.pool.sks-keyservers.net

  See the 2.2.17 section in the upstream NEWS file at
  /usr/share/doc/gnupg/NEWS.gz for more information about fully
  reverting to the old, risky behavior.
```

Their patch (includes more context): https://salsa.debian.org/debian/gnupg2/-/blob/debian/main/debian/patches/Use-hkps-keys.openpgp.org-as-the-default-keyserver.patch

**Comment by fanquake** [comment]
> Just a datapoint: it would seem that since the launch of keys.opengpg.org, many GPG tools and certain distros (Debian, NixOS) have switched over to that as the default server.

Looks like `keys.opengpg.org` also has more of the builder keys available than the Ubuntu server, as well as some newer signatures for keys that are available on either.

I've changed this PR to swap both the README and the CI to use `keys.opengpg.org`.

File: contrib/builder-keys/README.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ To fetch keys of builders and active developers, feed the list of fingerprints
 of the primary keys into gpg:
 
 ```sh
-while read fingerprint keyholder_name; do gpg --keyserver hkp://subset.pool.sks-keyservers.net --recv-keys ${fingerprint}; done < ./keys.txt
+while read fingerprint keyholder_name; do gpg --keyserver hkps://keys.openpgp.org --recv-keys ${fingerprint}; done < ./keys.txt
```
Comment (MarcoFalke) [inline contrib/builder-keys/README.md:23]: Are all builder keys on that keyserver? IIRC only persons with access to the email that the key is attached to can upload there.

Also the `src/minisketch/test` file added in this pull seems unrelated.

Comment (MarcoFalke) [reply contrib/builder-keys/README.md:23]: Nvm, just read your previous comment:

> Looks like keys.opengpg.org also has more of the builder keys available than the Ubuntu server, as well as some newer signatures for keys that are available on either.

**Comment by MarcoFalke** [comment]
Approach ACK b4f1f998236ab1ae44a1a3c8ae84213474cae4dc, but the exe should be removed

**Comment by fanquake** [comment]
>  but the exe should be removed

Done. Not sure how I didn't notice that. Fixed the commit message typo as well.

**Comment by MarcoFalke** [comment]
cr ACK b4f1f998236ab1ae44a1a3c8ae84213474cae4dc

**Comment by laanwj** [comment]
> hkps://hkps.pool.sks-keyservers.net is essentially no-longer functional,

It kind of gives me the feeling that we keep migrating from keyserver to keyserver as they become no-longer functional one by one.

But sure if it's needed it's needed. I don't see any better solution to this mess, mind you.

**Comment by laanwj** [comment]
FWIW, there's another mention of keyserver in `contrib/verify-commits/README.md`. Maybe replace this one too?

**Comment by Zero-1729** [comment]
Yeah, I think `contrib/verify-commits/README.md` might as well be updated too. More generally, though, I see moving to `keys.opengpg.org` as more of an LTS move; I don't think the keyserver will be down anytime soon.

**Comment by fanquake** [comment]
> FWIW, there's another mention of keyserver in contrib/verify-commits/README.md. Maybe replace this one too?

Thanks. Have replaced that as well.

**Comment by 04487** [comment]
در تاریخ سه‌شنبه ۱۷ اوت ۲۰۲۱،‏ ۰۵:۳۳ Michael Ford ***@***.***>
نوشت:

**Comment by MarcoFalke** [comment]
cr ACK 4c43b7d41d11072f382f938379d21cd2e0bcbb47

**Review by Zero-1729 — COMMENTED** [review]
ACK 4c43b7d41d11072f382f938379d21cd2e0bcbb47

**Comment by fanquake** [comment]
This was merged in fdd80b0a53b4af0b29cb6e03118e2456d053a757.
